keps/sig-node/5067-pod-generation/README.md

```diff
@@ -0,0 +1,1066 @@
+to avoid confusion. 
+
+Likewise, a new optional `observedGeneration` field will be added to the pod's 
+`status.condition` struct. This is to keep parity with the 
+[metav1.Condition struct](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L1620-L1625). 
+
+The net result will be a new `status.observedGeneration` field for the kubelet
+to express which generation the top-level status relates to, and a new `status.conditions[i].observedGeneration`
+field for the writer of that condition to express which generation that condition
+relates to.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+Because there is only one singular `status.observedGeneration` in the pod status, only
+one writer can set it. This is consistent with other object types that have
+`status.observedGeneration`, and the expectation is that the primary controller for
+the object sets the field. For a pod, that primary controller is the kubelet, so
+once a pod is bound to a node, we expect that the kubelet on that node is the
+sole writer of `status.observedGeneration`. 
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+#### Custom-set `metadata.generation`
+
+Today, it is possible for a client to custom-set a pod's `metadata.generation` on creation, but once
+set, the `metadata.generation` cannot be updated. That means that there is a
+possibility for an external client to be setting `metadata.generation` on pod 
+create and depend on that fixed value somehow in its own reconciliation logic.
+
+That said, the `metadata.generation` field is described as ["set by the system and monotonically increasing"](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#metadata),
+and thus it should be clear enough that `metadata.generation` was not intended to be
+used in this way.
+
+#### Infinite loop caused by misbehaving mutating webhooks
+
+It is possible that today there exists mutating webhooks that overwrite a pod's
+status. These older webhooks would not know about `status.observedGeneration`
+and could be clearing it. This would cause an infinite loop of the kubelet
+attempting to update a pod's `status.observedGeneration` and the webhook
+clearing it.
+
+Status-mutating webhooks could break more pod features than just what is
+proposed in this KEP, so we will not attempt to solve this here.
```

Comment:
I think we should mitigate this risk by improving docs: https://github.com/kubernetes/website/pull/46798#discussion_r1946911968

Reply:
sounds good! added this to the kep

Reply:
sounds good! added a note to the kep